src/runtime/mod.rs

```diff
@@ -227,6 +227,55 @@ impl Spec {
     }
 }
 
+impl From<Spec> for SpecBuilder {
+    fn from(spec: Spec) -> Self {
+        let mut spec_builder = SpecBuilder::default().version(spec.version());
+
+        spec_builder = match spec.root() {
```

Comment:
Can we use a macro to implement these logic?
I don't know if Rust can list all the attributes as a list or something, then we use a macro for these...
Maybe need some tricks.

Reply:
If we can do in this way, it's convenient to add this for other structs need similar From.

Reply:
I am not sure. I am open to ideas. I was hoping this as a short term stop gap. Long term, we should wait for https://github.com/colin-kiegel/rust-derive-builder/pull/215 to be merged. That would be proper solution in my opinion.

Reply:
Is it difficult to remove the `clone()`? I have a feeling that this is going to be a very resource-intensive process.

Reply:
Unfortunately I believe so. Yes, this is almost equivalent of making a new copy of `Spec` and will be more expensive. This is the price we pay to maintain immutability. The better method is likely `to_owned()`, but I think it will just be a clone in this case as well. The point is that this is a rare operation to invoke.